modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java

```diff
@@ -564,16 +569,7 @@ private CompletableFuture<AsyncSqlCursor<InternalSqlRow>> querySingle(
             validateParsedStatement(properties, result);
             validateDynamicParameters(result.dynamicParamsCount(), params, true);
 
-            QueryTransactionWrapper txWrapper = txCtx.getOrStartImplicit(result.queryType());
-
-            InternalTransaction tx = txWrapper.unwrap();
-
-            // Adding inflights only for read-only transactions.
-            if (tx.isReadOnly() && !transactionInflights.addInflight(tx.id(), tx.isReadOnly())) {
```

Comment:
this will be removed in https://issues.apache.org/jira/browse/IGNITE-21868 anyway